sycamore/transforms/query.py

```diff
@@ -0,0 +1,57 @@
+from abc import abstractmethod, ABC
+from typing import Any
+
+from opensearchpy import OpenSearch
+from ray.data import Dataset
+
+from sycamore.data import OpenSearchQueryResult, Element, OpenSearchQuery
+from sycamore.plan_nodes import Node, NonCPUUser, NonGPUUser, Transform
+from sycamore.utils.generate_ray_func import generate_map_function
+
+
+class QueryExecutor(ABC):
```

Comment:
Trying to understand when we would have another executor. Aren't the query objects kind of OpenSearch specific?

Reply:
I expect we'll want a simplified query interface (e.g. just send in a question instead of os specifics). I've also just been building all classes with the assumption of having a different db